3.analyze_data/UMAP_analysis/nbconverted/2.jump_UMAP.py

```diff
@@ -0,0 +1,389 @@
+
+print(mito_jump_zernike_df.shape)
+mito_jump_zernike_df.head(2)
+
+
+# ## Only AreaShape features
+
+# In[12]:
+
+
+# Extract metadata columns
+metadata_cols = [
+    col for col in merged_mito_jump_df.columns if col.startswith("Metadata")
+]
+# Extract feature columns
+feature_cols = [col for col in merged_mito_jump_df.columns if col.startswith("Nuclei")]
+
+# Filter feature columns for "Zernike"
+areashape_feature_cols = [col for col in feature_cols if "AreaShape" in col]
+
+# Create a new DataFrame with metadata and filtered features
+mito_jump_areashape_df = merged_mito_jump_df[metadata_cols + areashape_feature_cols]
+
+# Add df to dictionary
+df_dict["Mito_JUMP_areashape_features"] = mito_jump_areashape_df
+
+print(mito_jump_areashape_df.shape)
+mito_jump_areashape_df.head(2)
+
+
+# ## Generate UMAP coordinates for each of the data splits
+# 
+# 1. All Nuclei features from JUMP
+# 2. All Nuclei features from Mitocheck + JUMP
+# 3. Only Zernike features (AreaShape measurement) from Mitocheck + JUMP
+# 4. Only AreaShape features from Mitocheck + JUMP
+
+# In[13]:
+
+
+# Set constants
+umap_random_seed = 0
+umap_n_components = 2
+
+for data_name, df in df_dict.items():
+    print(
+        "Creating embeddings for",
+        data_name,
+        "and including the probabilities from the",
+        model_type,
+        "_".join(prob_path.stem.split("_")[:2]),
+        "model",
+    )
+    # Make sure to reinitialize UMAP instance per plate
+    umap_fit = umap.UMAP(random_state=umap_random_seed, n_components=umap_n_components)
+
+    # Process df to separate features and metadata
+    metadata_columns = [col for col in df.columns if col.startswith("Metadata")]
+    feature_columns = [col for col in df.columns if not col.startswith("Metadata")]
+
+    # Fit UMAP and convert to pandas DataFrame
+    embeddings = pd.DataFrame(
+        umap_fit.fit_transform(df.loc[:, feature_columns]),
+        columns=[f"UMAP{x}" for x in range(0, umap_n_components)],
+    )
+
+    # Combine with metadata
+    umap_with_metadata_df = pd.concat([df.loc[:, metadata_columns], embeddings], axis=1)
+
+    if not data_name == "Only_JUMP_all_features":
+        # Only include relevant metadata and UMAP coords (merged only)
+        umap_with_metadata_df = umap_with_metadata_df[
+            [
+                "Metadata_data_name",
+                "Metadata_Predicted_Class",
+                "Metadata_Phenotypic_Value",
+                "UMAP0",
+                "UMAP1",
+            ]
+        ]
+
+    # Make folder per data split
+    model_dir = pathlib.Path(f"./{UMAP_results_dir}/{data_name}")
+    model_dir.mkdir(exist_ok=True)
+
+    # Generate output file and save
+    output_umap_file = pathlib.Path(
+        f"./{model_dir}/{data_name}_{model_type}_{'_'.join(prob_path.stem.split('_')[:2])}_model.tsv"
+    )
+    umap_with_metadata_df.to_csv(output_umap_file, index=False, sep="\t")
+
+print(umap_with_metadata_df.shape)
+umap_with_metadata_df.head()
```

Comment:
Great! So just to confirm, all UMAPs you describe here will have output coordinates with Metadata so that I can create figures in the phenotypic_profiling repo?

Reply:
Correct! All of these will have UMAP coordinates and associated in Metadata expect in the Mito+JUMP data split, the only Metadata is the predicted class, predicted value, and data set since the JUMP Metadata didn't match with Mito.